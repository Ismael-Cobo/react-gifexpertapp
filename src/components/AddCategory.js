import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setcategories}) => {
    
    const [inputValue, setinputValue] = useState('') 
    // ponerle las comillas al useState porque si no será siempre un undefined
    // y cuandoi intentes cambiar el valor petará                                                

    const handleInputValue = (e) => {
        // Cada vez que cambies el valor del input lo has de modificar 
        // en el useState
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length > 2){
            setcategories(category => [inputValue,...category])
            setinputValue('')
            console.log('Submit hecho')
        }
    }

    return (
        
        <form onSubmit = {handleSubmit}>
            <input 
                type = "text"
                value = { inputValue }
                onChange = { handleInputValue }
            />
        </form>
        
    )
}


AddCategory.protoTypes = {
    setcategories : PropTypes.func.isRequired
}

