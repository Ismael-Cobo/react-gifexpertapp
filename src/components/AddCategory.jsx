import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')
    
    const handleInputCahnge = ({target}) => {
        setInputValue(target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length > 2) {
            setCategories( category => [inputValue, ...category])
            setInputValue('')
        }
    }

    return (
        <form 
            className='gif__aÃ±adir_categoria'
            onSubmit={handleSubmit}
        >
            <input type="text" value={inputValue} onChange={handleInputCahnge}/>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}