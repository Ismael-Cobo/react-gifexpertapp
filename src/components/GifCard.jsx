import PropTypes from 'prop-types'

export const GifCard = ({url, title}) => {
    return (
        <div className="card">
            <div className="card__img">
                <img src={url} alt={title} />
            </div>
            <p>{title}</p>
        </div>
    )
}

GifCard.propTypes = {
    url: PropTypes.string,
    title: PropTypes.string
}