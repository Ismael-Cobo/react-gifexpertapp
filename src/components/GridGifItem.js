import React from 'react'

export const GridGifItem = ( { id, title, url } ) => {

    return (
            <div className='card'>
                    <div className='card__img'>
                        <img src={ url } alt={title} />
                    </div>
                
                <p>{ title }</p>
            </div>
    )
}
