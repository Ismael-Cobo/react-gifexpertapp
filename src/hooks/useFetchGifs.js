import { useEffect, useState } from "react"
import { fetchGifs } from "../helpers/fetchGifs"

export const useFetchGifs = (category) => {

    const [data, setData] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {

        fetchGifs(category)
            .then(resp => {
                setData({
                    data: [...resp],
                    loading: false
                })
                
            })

    }, [category])

    return data
}