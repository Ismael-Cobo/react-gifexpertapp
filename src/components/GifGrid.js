import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs'
import { GridGifItem } from './GridGifItem'

export const GifGrid = ({ category }) => {

    const {data: images, loading} = useFetchGifs(category);
    
    /*const [images, setimages] = useState([])

    useEffect( () => {
        getGifs(category)
            .then(imgs => setimages(imgs))
    }, [category])

    */

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>loading...</p> }

            <div className='card-grid' >
            
                {
                    images.map( img => (
                        <GridGifItem 
                            key = {img.id} 
                            { ...img }
                        />
                    ))
                }
            
            </div>  
            
        </>
    )
}
