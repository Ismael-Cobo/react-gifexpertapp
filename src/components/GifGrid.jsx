import PropTypes from 'prop-types'
import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifCard } from "./GifCard"

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs(category)
    
    return (
        <>
            <h3>{category}</h3>

            <div className="card-grid">
                {/* Aqui mapeo los items */}
                {loading && <h4>Loading...</h4>}
                {
                    data.map(img => (

                        <GifCard
                            key={img.id}
                            url={img.url}
                            title={img.title}
                        />

                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string
}