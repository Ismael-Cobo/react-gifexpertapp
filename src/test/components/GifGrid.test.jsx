import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe('prueba de <GifGrid />', () => {

    const category = 'One punch'

    beforeEach(() => {
        jest.clearAllMocks()
    })

    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de cambiar la snapshot cuando el useFetchGifs se ejecute', () => {

        const gifs =[{
            id: 123,
            title: 'Imagen',
            url: 'https://localhost:3000/img.jpg'
        },
        {
            id: 456,
            title: 'Imagen2',
            url: 'https://localhost:3000/img2.jpg'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />)
        
        //expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifCard').length).toBe(gifs.length)

    })

})