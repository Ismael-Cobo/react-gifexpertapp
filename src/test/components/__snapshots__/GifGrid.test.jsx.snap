// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba de <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <h4>
      Loading...
    </h4>
  </div>
</Fragment>
`;
